addons/sys_data/script_component.hpp

```diff
@@ -14,4 +14,6 @@
     #define DEBUG_SETTINGS DEBUG_SETTINGS_SYS_DATA
 #endif
 
+#define IS_SERIALIZEDHASH(array) (IS_ARRAY(array) && {(count array) > 0} && {IS_STRING((array select 0))} && {(array select 0) == "ACRE_HASH"})
```

Comment:
That still doesn't guarantee the variable  `array` is of the type array?
Also `isEqualTo` will be case-sensitive (unsure if it matters in this case)
So there would certainly be cases where you suggested optimisation is not a good idea.